src/common.c

```diff
@@ -64,7 +64,21 @@ int fi_poll_fd(int fd, int timeout)
 
 	fds.fd = fd;
 	fds.events = POLLIN;
-	ret = poll(&fds, 1, timeout);
+
+	// set the timeout value
+	struct timespec tmo = {0, timeout*1000000 /* milisec -> nanosec */};
+
+	// to set the signal mask
+        sigset_t set;
+
+         /* Block SIGUSR1, SIGUSR2, and all realtime signals from SIGRTMIN to SIGRTMAX */
+        sigemptyset(&set);
+        sigaddset(&set, SIGUSR1);
+        sigaddset(&set, SIGUSR2);
+	for (int i = SIGRTMIN; i <= SIGRTMAX; i++)
+	        sigaddset(&set, i);
+
+	ret = ppoll(&fds, 1, &tmo, &set);
```

Comment:
This is a Linux specific call that breaks OS X, Free BSD, and Windows.  Unless there's a portable solution that can also work with a broad set of apps, apps that want to catch signals may need to handle this internally, rather than relying on libfabric catching this as part of internally blocking on an fd.